floris/tools/optimization/scipy/yaw.py

```diff
@@ -289,10 +299,26 @@ def reinitialize_opt(
                 turbine.yaw_angle
                 for turbine in self.fi.floris.farm.turbine_map.turbines
             ]
+        self.x0_norm = self._norm(
+            np.array(self.x0), self.minimum_yaw_angle, self.maximum_yaw_angle
+        )
         if bnds is not None:
             self.bnds = bnds
+            self.minimum_yaw_angle = np.min([bnds[i][0] for i in range(self.nturbs)])
+            self.maximum_yaw_angle = np.max([bnds[i][1] for i in range(self.nturbs)])
         else:
             self._set_opt_bounds(self.minimum_yaw_angle, self.maximum_yaw_angle)
+        self.bnds_norm = [
+            (
+                self._norm(
+                    self.bnds[i][0], self.minimum_yaw_angle, self.maximum_yaw_angle
+                ),
+                self._norm(
+                    self.bnds[i][1], self.minimum_yaw_angle, self.maximum_yaw_angle
+                ),
+            )
+            for i in range(self.nturbs)
+        ]
```

Comment:
I changed how the bounds are set and normalized if `bnds` is defined in the arguments. Before, `bnds_norm` was getting set to (0,1) for each turbine, even if `bnds` specified different bounds for each turbine.